src/ark/utils/deepcell_service_utils.py

```diff
@@ -99,85 +176,41 @@ def create_deepcell_output(deepcell_input_dir, deepcell_output_dir, fovs=None,
-
-        # sort by newest added
-        zip_files.sort(key=os.path.getmtime)
-
-        with ZipFile(zip_files[-1], "r") as zipObj:
-            for name in zipObj.namelist():
-                # this files will only ever be suffixed with feature_0.tiff or feature_1.tiff
-                if '_feature_0.tif' in name:
-                    resuffixed_name = name.replace('_feature_0', wc_suffix)
-                else:
-                    resuffixed_name = name.replace('_feature_1', nuc_suffix)
-
-                mask_path = os.path.join(deepcell_output_dir, resuffixed_name)
-
-                # DeepCell uses .tif extension, append extra f to account for .tiff standard
-                mask_path += 'f'
-
-                # read the file from the .zip file and save as segmentation mask
-                byte_repr = zipObj.read(name)
-                ranked_segmentation_mask = (_convert_deepcell_seg_masks(byte_repr)).squeeze()
-                image_utils.save_image(mask_path, ranked_segmentation_mask)
+            # pass the zip file to deepcell.org
+            status = run_deepcell_direct(
+                input_zip_path, deepcell_output_dir, host, job_type, scale, timeout
+            )
 
-            # verify that all the files were extracted
-            for fov in fov_group:
-                if fov + '_feature_0.tif' not in zipObj.namelist():
-                    warnings.warn(f'Deep Cell whole cell output file was not found for {fov}.')
-                if fov + '_feature_1.tif' not in zipObj.namelist():
-                    warnings.warn(f'Deep Cell nuclear output file was not found for {fov}.')
+            # ensure execution is halted if run_deepcell_direct returned non-zero exit code
+            if status != 0:
+                print("The following FOVs could not be processed: %s" % ','.join(fov_group))
+                return
+            # successful output
+            else:
+                # extract segmentation masks from deepcell output
+                extract_deepcell_response(deepcell_output_dir, fov_group, batch_num, wc_suffix,
+                                          nuc_suffix)
 
-    # make calls in parallel
-    if parallel:
-        with ThreadPoolExecutor() as executor:
-            executor.map(_zip_run_extract, fov_groups, range(len(fov_groups)))
-            executor.shutdown(wait=True)
-    else:
-        list(map(_zip_run_extract, fov_groups, range(len(fov_groups))))
+    deepcell_outputs = io_utils.remove_file_extensions(
+        io_utils.list_files(deepcell_output_dir, substrs=[".zip"]))
+    misc_utils.verify_same_elements(
+        output_files=deepcell_outputs,
+        expected_files=[f"deepcell_response_fovs_batch_{i + 1}" for i in range(len(fov_groups))])
 
 
 def run_deepcell_direct(input_dir, output_dir, host='https://deepcell.org',
-                        job_type='mesmer', scale=1.0, timeout=3600, num_retries=5):
+                        job_type='mesmer', scale=1.0, timeout=3600):
```

Comment:
I'd say let's default the timeout to something smaller, 1 hour is a long time to wait for a single batch to upload. 90 or 120 is more reasonable.